kart/annotations/__init__.py

```diff
@@ -0,0 +1,70 @@
+import json
+import logging
+from .db import annotations_session, KartAnnotation
+
+L = logging.getLogger(__name__)
+
+
+class DiffAnnotations:
+    def __init__(self, repo):
+        self.repo = repo
+
+    def _object_id(self, base_rs, target_rs):
+        # this is actually symmetric, so we can marginally increase hit rate by sorting first
+        if base_rs.tree.id < target_rs.tree.id:
+            return f"{base_rs.tree.id}...{target_rs.tree.id}"
+        else:
+            return f"{target_rs.tree.id}...{base_rs.tree.id}"
```

Comment:
I know we use `(base|target)_rs` throughout diff, but what does it mean?

Reply:
```suggestion
        tree_ids = sorted(rs.tree.id for rs in (base_rs, target_rs))
        return f"{tree_ids[0]}...{tree_ids[1]}"
```

Reply:
Added notes to docstring about the repostructure objects